File: src/main/java/org/example/WeightTester.java
package org.example;

import java.util.Optional;

import static org.example.ValidateTester.validateFor;

public class WeightTester extends Tester {
    @Override
    protected boolean validate( Product product ) {
        return product.weight() >= 1200 && product.weight() <= 1300
                ? Optional.ofNullable( getNextTester() )
                          .map( validateFor( product ) )
                          .orElse( true )
                : false;
    }
}

File: src/main/kotlin/WeightTester.kt
import extensions.isInBetween

class WeightTester : Tester() {
    override fun test(product: Product): Boolean =
        product.validate()

    private fun Product.validate() =
        if (weight.isInBetween(1200, 1300))
            nextTester.test(this)
        else false
}